### ISSUE TITLE
Add MongoDB instrumentation

### ISSUE THREAD

**Issue by glucaci** [open-telemetry/opentelemetry-dotnet-contrib#127]
There is aleady an implementation from @jbogard.

https://github.com/jbogard/MongoDB.Driver.Core.Extensions.DiagnosticSources

If he would like, we have to see what is needed to integrate it here.
I would like to see this part of the official OpenTelemetry packages.

**Comment by lucasoares** [comment]
We started using MongoDB with .NET recently and this would be much appreciated. Having it as a official opentelemetry package would make it possible for us to start using it. @jbogard any chance on bringing https://github.com/jbogard/MongoDB.Driver.Core.Extensions.DiagnosticSources to openelemetry-dotnet-contrib?

**Comment by Kielek** [comment]
1. MongoDB team promises native support for this: https://jira.mongodb.org/browse/CSHARP-3124. No updates so far.
2. This package will be exactly the same official as the one created by the @jbogard. The only difference will be the namespace and the nuget-owners.
3. It will be great to agree with @jbogard here that the old package will be dropped. Without this we will have 2 packages, and it is the worst case.
4. Db semantic convention going to be stabilized shortly. It is a blocker to release stable version, you can expect only pre-releases.

**Comment by lucasoares** [comment]
> Clarification though - `opentelemetry-dotnet-contrib` does not make it "official" - that repo is just a collection of community-contributed packages. That's the "contrib" part of it. "Official" would be MongoDB support directly. Aspire's official MongoDB package already uses this package so I'm not sure what problems moving repos could possible solve here.
> 
> I would rather not move this over to a separate repo when this functionality should belong in the MongoDB driver itself. Perhaps the better solution here is to wait until the DB semantic convention to become stabilized then open a PR with the MongoDB driver repo itself?

Yeah, I totally agree with you. But company policies are company policies, right? HAHAHA I will wait for it then =)

**Comment by jbogard** [comment]
I'm not suggesting putting your job at risk, but I can't imagine how "this package is used in Microsoft's public packages" wouldn't make this "official". See: https://www.nuget.org/packages/Aspire.MongoDB.Driver

It's official enough for Microsoft and Aspire, but isn't official enough for your company? Hmmmm 🤣

**Comment by martincostello** [comment]
As noted above, existing packages exist, and there's a plan _at some point_ to add native instrumentation to the MongoDB driver.

We won't be adding a competing (and eventually obsolete) implementation here.

Nothing here is "official", it's just maintained under the OpenTelemetry umbrella by the maintainers here, but the code is "owned" by whichever community members want to own it (typically those that add it).

We're not going to increase our maintenance burden to satisfy arbitrary third-party company policies.

**Comment by jbogard** [comment]
FYI there is a PR open for providing this support directly in the MongoDB C# driver:

https://github.com/mongodb/mongo-csharp-driver/pull/1837

When this is merged/released, I'll mark my package as deprecated/obsolete. And throw a little party for one less OSS package I need to support!
